### PR TITLE
add device! functions for FixedAdmittance 

### PR DESCRIPTION
**Pull request by m-bossart**

### CODE REVIEW

File: src/models/load_models.jl [hunk 1]
```diff
@@ -22,3 +22,17 @@ function mdl_Zload!(voltage_r, voltage_i, current_r, current_i, device::PSY.Powe
...
+function mdl_Zload!(voltage_r, voltage_i, current_r, current_i, device::PSY.FixedAdmittance)
+    #TODO - implement the new load
+    bus = PSY.get_bus(device)
+
+    #Load device parameters
+    Y = PSY.get_Y(device)
+    G = real(Y)
+    B = imag(Y)
+
+    current_r[1] += -(voltage_r[1] * G + voltage_i[1] * B)  #in system pu flowing out
```
Comment (rodrigomha) [inline src/models/load_models.jl:35]: This does not seem right:
From what I understand, the current going into the admittance can be computed as:
I = -YV = -(G+jB)(Vr+jVi) = -(GVr - BVi) - j(GVi + BVr)
The term BVi seems to be with a different sign.

Also we need a test using a Fixed Admittance to pass the codecov/patch requirement.

**Review by rodrigomha — CHANGES_REQUESTED** [review]

File: src/models/load_models.jl (hunk 1 above)
Comment (jd-lara) [reply src/models/load_models.jl:35]: @rodrigomha correct, the formula needs to be modified.

**Comment by m-bossart** [comment]
Good catch, does the test need to include a comparison to psse/pscad or just build and solve a system with a fixed admittance?

**Comment by jd-lara** [comment]
> Good catch, does the test need to include a comparison to psse/pscad or just build and solve a system with a fixed admittance?

For now just make a test that runs a case with the FixedAdmittance we can compare with pss/e later

**Comment by m-bossart** [comment]
I'm seeing some behavior that I can't explain when solving the power flow for a system with a FixedAdmittance. I'm using the version with the previous typo corrected. I thought that the two display statements would give the same bus voltages but the solution is different. My understanding is that the `Simulation!()` call would solve the power flow and update the system voltages and power references and the `solve_powerflow()` call would give the same result. Any thoughts on what the issue could be? When I remove the FixedAdmittance the results are the same. 
```
using Pkg
Pkg.activate(".")
using Revise
using OrdinaryDiffEq
using PowerSystems
using PowerSimulationsDynamics
include("../models/DynamicComponents.jl")

sys = System("cases/IEEE 14 bus_modified_33.raw")
#sys = System("cases/IEEE 14 bus_modified_33_RemoveFixedAdmittance.raw")

for g in get_components(ThermalStandard,sys)
    inv_case = inv_case78(get_name(g))
    add_component!(sys, inv_case, g)
end

sim = Simulation!(MassMatrixModel, sys, pwd(), (0.0,1.0))
display(solve_powerflow(sys)["bus_results"])
display(get_initial_conditions(sim)["Vm"])
```

**Comment by rodrigomha** [comment]
> I'm seeing some behavior that I can't explain when solving the power flow for a system with a FixedAdmittance. I'm using the version with the previous typo corrected. I thought that the two display statements would give the same bus voltages but the solution is different. My understanding is that the `Simulation!()` call would solve the power flow and update the system voltages and power references and the `solve_powerflow()` call would give the same result. Any thoughts on what the issue could be? When I remove the FixedAdmittance the results are the same.
> 
> ```
> using Pkg
> Pkg.activate(".")
> using Revise
> using OrdinaryDiffEq
> using PowerSystems
> using PowerSimulationsDynamics
> include("../models/DynamicComponents.jl")
> 
> sys = System("cases/IEEE 14 bus_modified_33.raw")
> #sys = System("cases/IEEE 14 bus_modified_33_RemoveFixedAdmittance.raw")
> 
> for g in get_components(ThermalStandard,sys)
>     inv_case = inv_case78(get_name(g))
>     add_component!(sys, inv_case, g)
> end
> 
> sim = Simulation!(MassMatrixModel, sys, pwd(), (0.0,1.0))
> display(solve_powerflow(sys)["bus_results"])
> display(get_initial_conditions(sim)["Vm"])
> ```

Okay, I have an idea. Maybe everything we are doing is not necessary. Can you check if the Ybus is different with/without the FixedAdmittance? If is different, then we don't need the add_device function, since the Ybus is used directly in the Kirchoff Law equations to compute the flow into the bus.

**Comment by jd-lara** [comment]
> I'm seeing some behavior that I can't explain when solving the power flow for a system with a FixedAdmittance. I'm using the version with the previous typo corrected. I thought that the two display statements would give the same bus voltages but the solution is different. My understanding is that the `Simulation!()` call would solve the power flow and update the system voltages and power references and the `solve_powerflow()` call would give the same result. Any thoughts on what the issue could be? When I remove the FixedAdmittance the results are the same.
> 
> ```
> using Pkg
> Pkg.activate(".")
> using Revise
> using OrdinaryDiffEq
> using PowerSystems
> using PowerSimulationsDynamics
> include("../models/DynamicComponents.jl")
> 
> sys = System("cases/IEEE 14 bus_modified_33.raw")
> #sys = System("cases/IEEE 14 bus_modified_33_RemoveFixedAdmittance.raw")
> 
> for g in get_components(ThermalStandard,sys)
>     inv_case = inv_case78(get_name(g))
>     add_component!(sys, inv_case, g)
> end
> 
> sim = Simulation!(MassMatrixModel, sys, pwd(), (0.0,1.0))
> display(solve_powerflow(sys)["bus_results"])
> display(get_initial_conditions(sim)["Vm"])
> ```

It could be the admittance vs power modeling between them. I think the power flow assumes that the fixed admittance

**Comment by m-bossart** [comment]
@rodrigomha The Ybus is different, so I think you are right that we do not need add_device for the FixedAdmittance

**Comment by jd-lara** [comment]
@m-bossart if you think this is not needed we can close the PR and revisit this later if we want to model the loads differently

**Pull request closed without merging**